### PR TITLE
updating dex container image

### LINKED ISSUE #800: Dex container to update in manifest
I'm using Dex authentication, with authentication delegated to another OIDC IDP.

In the configmaps of dex, I configured `getUserInfo: true` to make dex retrieve the claim 'name' but it fails with the message `Failed to authenticate: missing "name" claim`. 

The current installation of dex is pointing to the container gcr.io/arrikto/dexidp/dex:4bede5eb80822fc3a7fc9edca0ed2605cd339d17 (in the file dex-auth/dex-crd/base/kustomization.yaml); for which I was not able to retrieve the corresponding Dockerfile.
I made my own overlay to use `quay.io/dexidp/dex:v2.21.0` and it works perfectly.

Would it be possible to get some details about the reason why this specific container was used instead of the standard dex container ? Maybe I will face some issues I'm not aware of. :-(

Maybe it could be a good idea to switch back to the standard dex container.

### PR DESCRIPTION
**Pull request by WillBeebe**
**Which issue is resolved by this Pull Request:**
Resolves #800 

**Description of your changes:**
This switches Dex back to the standard container. `quay.io/coreos/dex` is [no longer](https://quay.io/repository/coreos/dex?tab=tags) maintained. Looking at the [releases](https://github.com/dexidp/dex/releases) the new image repo is `quay.io/dexidp` thus the change.

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/813)

### CODE REVIEW

File: dex-auth/dex-crds/base/kustomization.yaml [hunk 1]
```diff
@@ -78,7 +78,3 @@ vars:
     fieldpath: data.application_secret
 configurations:
 - params.yaml
-images:
```
Comment (krishnadurai) [inline dex-auth/dex-crds/base/kustomization.yaml]: @WillBeebe Its recommended that we retain the 'images' field in kustomization.yaml. Could you please retain this field and update it with the image name.
This allows us to parameterize images through CLI.

Comment (WillBeebe) [reply dex-auth/dex-crds/base/kustomization.yaml]: No problem, that makes sense. :) Just pushed the change.

**Comment by krishnadurai** [comment]
@WillBeebe could you please rebase?

**Comment by WillBeebe** [comment]
@krishnadurai done and done

**Comment by krishnadurai** [comment]
@WillBeebe the latest image is v2.22.0. Could you please use this image instead?

**Comment by WillBeebe** [comment]
@krishnadurai updated. :)

**Comment by krishnadurai** [comment]
/lgtm
/approve
/hold

@WillBeebe You can remove hold when tests pass.

**Comment by WillBeebe** [comment]
Hi @krishnadurai I don't have the permission to remove labels on this repo.

**Pull request merged**